jax/tree_util.py

```diff
@@ -48,17 +49,23 @@
 traceback_util.register_exclusion(__file__)
 
 
-def tree_flatten(tree):
+def tree_flatten(tree, is_leaf: Optional[Callable[[Any], bool]] = None):
   """Flattens a pytree.
 
   Args:
     tree: a pytree to flatten.
+    is_leaf: an optionally specified function that will be called at each
+      flattening step. It should return a boolean, which indicates whether
+      the flattening should traverse the current object, or if it should be
+      stopped immediately, with the whole subtree being treated as a leaf.
 
   Returns:
     A pair where the first element is a list of leaf values and the second
     element is a treedef representing the structure of the flattened tree.
   """
-  return pytree.flatten(tree)
+  # We skip the second argument in support of old jaxlibs
```

Comment:
Leave a TODO comment saying which jaxlib version must be the minimum to allow us to remove the fallback case.